src/pages/physical-count-itemDetails/index.js

```diff
@@ -0,0 +1,633 @@
+                  formik.setFieldValue('stockCountId', newValue?.recordId)
+                  formik.setFieldValue('siteId', '')
+
+                  if (!newValue) {
+                    setSiteStore([])
+                    setFilteredItems([])
+                    clearGrid()
+                    formik.setFieldValue('SCStatus', null)
+                    formik.setFieldValue('SCWIP', null)
+
+                    //setDisableItemDuplicate(false)
+                  } else {
+                    fillSiteStore(newValue?.recordId)
+                    formik.setFieldValue('SCStatus', newValue?.status)
+                    formik.setFieldValue('SCWIP', newValue?.wip)
+                  }
+
+                  setDisableItemDuplicate(!!newValue?.disableItemDuplicate)
+                  console.log('bool', !!newValue?.disableItemDuplicate)
+                }}
+                error={formik.touched.stockCountId && Boolean(formik.errors.stockCountId)}
+              />
+            </Grid>
+            <Grid item xs={2}>
+              <ResourceComboBox
+                name='siteId'
+                store={siteStore}
+                label={_labels.site}
+                valueField='siteId'
+                displayField={['siteRef', 'siteName']}
+                columnsInDropDown={[
+                  { key: 'siteRef', value: 'Reference' },
+                  { key: 'siteName', value: 'Name' }
+                ]}
+                values={formik.values}
+                required
+                readOnly={formik.values.controllerId}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('siteId', newValue?.siteId)
+                  formik.setFieldValue('controllerId', '')
+
+                  if (!newValue) {
+                    setControllerStore([])
+                    setFilteredItems([])
+                    clearGrid()
+                    formik.setFieldValue('EndofSiteStatus', null)
+                  } else {
+                    fillControllerStore(formik.values.stockCountId, newValue?.siteId)
+                    formik.setFieldValue('EndofSiteStatus', newValue?.status)
+                  }
+                }}
+                error={formik.touched.siteId && Boolean(formik.errors.siteId)}
+                maxAccess={maxAccess}
+              />
+            </Grid>
+            <Grid item xs={2}>
+              <ResourceComboBox
+                name='controllerId'
+                store={controllerStore}
+                label={_labels.controller}
+                valueField='controllerId'
+                displayField='controllerName'
+                values={formik.values}
+                required
+                readOnly={formik.values.controllerId}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('controllerId', newValue?.controllerId)
+                  checkPhyStatus(formik.values.stockCountId, formik.values.siteId, newValue?.controllerId)
```

Comment:
Both checkPhyStatus and fetchGridData functions can take only controllerId as parameter to minimize delays. The stockCountId and siteId can be accessed directly from formik within each function, eliminating the need to pass them as parameters.